pkgs/development/libraries/dlib/default.nix

```diff
@@ -33,20 +35,37 @@ stdenv.mkDerivation rec {
   '';
 
   cmakeFlags = [
-    (lib.cmakeBool "USE_DLIB_USE_CUDA" cudaSupport)
     (lib.cmakeBool "USE_SSE4_INSTRUCTIONS" sse4Support)
     (lib.cmakeBool "USE_AVX_INSTRUCTIONS" avxSupport)
+    (lib.cmakeBool "DLIB_USE_CUDA" cudaSupport)
+  ] ++ lib.optionals cudaSupport [
+    (lib.cmakeFeature "DLIB_USE_CUDA_COMPUTE_CAPABILITIES" (builtins.concatStringsSep "," (with cudaPackages.flags; map dropDot cudaCapabilities)))
   ];
 
-  nativeBuildInputs = [ cmake pkg-config ];
+  nativeBuildInputs = [ cmake pkg-config ] ++ lib.optional cudaSupport cudaPackages.cuda_nvcc;
 
   buildInputs = [
     fftw
     libpng
     libjpeg
     libwebp
     openblas
-  ] ++ lib.optional guiSupport libX11;
+  ]
+  ++ lib.optional guiSupport libX11
+  ++ lib.optionals config.cudaSupport (with cudaPackages; [
+    cuda_cudart
+    cuda_nvcc.dev # crt/host_config.h
+    libcublas
+    libcurand
+    libcusolver
```

Comment:
If you'd like to have the smaller runtime closures right now you might want to consider using explicit outputs: `.lib`, `.dev`, `.static`.

This is because of https://github.com/NixOS/nixpkgs/issues/